scalyr_agent/util.py

```diff
@@ -380,6 +380,23 @@ def create_unique_id():
     return base64_id.decode("utf-8")
 
 
+def create_uuid3(namespace, name):
+    """
+    Return new UUID based on a hash of a UUID namespace and a string.
+    :param namespace: The namespace
+    :param name: The string
+    :type namespace: uuid.UUID
+    :type name: six.text
+    :return:
+    :rtype: uuid.UUID
+    """
+    if six.PY2:
+        return uuid.uuid3(namespace, six.ensure_binary(name))
+    else:
+        # noinspection PyTypeChecker
+        return uuid.uuid3(namespace, six.ensure_text(name))
```

Comment:
I think we can avoid using `if` statement by using `six.ensure_str` https://six.readthedocs.io/#six.ensure_str

```suggestion
    return uuid.uuid3(namespace, six.ensure_str(name))
```

Reply:
Yep what Arthur said.

Besides that, LGTM.